### ISSUE TITLE
mathJax stop working on Windows Phone 8.1 after newest update

### ISSUE THREAD

**Issue by Sebus13** [mathjax/MathJax#1085]
Hi,
I have an app that was using MathJax on Lumia 820 with Windows Phone 8.1. Everything was working great but after the last one update which I made to Windows Phone 8.1 Update 1, OS Version: 8.10.14234.375 MathJax stop working in my app. I checked that also when I run Internet Explorer on my Lumia 820 and go to https://www.mathjax.org/ page MathJax does not render equations. I checked it on a second phone (Lumia 535) with the same os version and it also does not render.
I suppose that MathJax stop working on all wp 8.1. phones with this os version

**Comment by pkra** [comment]
Thanks for filing this here, too. I don't have any means to access Windows Phone 8.1 -- does anybody have pointers to web-services or VMs?

**Comment by bamorim** [comment]
Some of my customers are issuing the same problem. My roommate has recently bought a windows phone. I'll try to investigate it too.

**Comment by bamorim** [comment]
One customer reported that it was not working in any browser (on windows phone). He tried IE, Opera(I think) and Chrome.

**Comment by pkra** [comment]
It would help to get more information. Does this problem occur on the examples at http://cdn.mathjax.org/mathjax/latest/test/examples.html ? (This might be a problem particular to the demos on mathjax.org rather than MathJax.)

**Comment by pkra** [comment]
I tested mathjax.org with the Win8 Phone emulator and couldn't detect any serious problems (cf screeshot).

![zc870fe8d152f93b9535](https://cloud.githubusercontent.com/assets/426686/7734113/0ef0f23a-ff34-11e4-85ae-2a97b3513f30.png)

Not sure what to do at this point.

**Comment by iver56** [comment]
When testing on my Lumia 820 with Windows 8.1, this page renders correctly:
http://cdn.mathjax.org/mathjax/latest/test/sample-tex.html
but this page does not render correctly:
http://cdn.mathjax.org/mathjax/latest/test/sample-mml.html
However, the first render (during the loading) looks kinda ok until it is done rendering. The same thing happens when I look at www.mathjax.org. The first few seconds I can see the fraction rendered "correctly", but after a few seconds, when the final rendering is done, it looks broken.

**Comment by Sebus13** [comment]
Hi,
I also tested on my Lumia 820 with Windows Phone 8.1 and can confirm that sample-tex renders correctly:
![wp_ss_20150524_0001](https://cloud.githubusercontent.com/assets/11392432/7787748/c951f51c-021c-11e5-8d63-eeca5e9d2476.png)
but sample-mml doesn't renders correctly:
![wp_ss_20150524_0002](https://cloud.githubusercontent.com/assets/11392432/7787751/eb96e8f8-021c-11e5-8ebd-c1432ec2976d.png)
I also can confirm that when I go to www.mathjax.org the intial render (what we're seeing while the page is still being loaded) looks kinda ok:
![wp_ss_20150524_0005](https://cloud.githubusercontent.com/assets/11392432/7787759/215052cc-021d-11e5-83f4-170981950406.png)
until it is done rendering:
![wp_ss_20150524_0006](https://cloud.githubusercontent.com/assets/11392432/7787762/37379ad2-021d-11e5-81f7-21953b395f27.png)
I can also confirm that everything works ok when I switch the phone's browser to desktop mode.

Let me know if there's anything else I can test/do.

**Comment by pkra** [comment]
Thanks so much for the additional feedback, @Sebus13  @iver56! I've updated the label of the issue.

FYI on mathjax.org the equation at the top is actually a pre-generated SVG (via MathJax-node) so this is not related to the issue.

@dpvc any suggestions?

**Comment by iver56** [comment]
I guess the problem is that the Windows phone is being misidentified as a mobile device that runs Android or iOS, and as a result MathJax goes into a mode that doesn't work in IE mobile.

**Comment by pkra** [comment]
@iver56 might be. But we noticed a similar issue with Spartan/Edge recently which came down to a bug in Edge wrt to handling `<math>` tags (and that one has since been fixed in Edge).

**Comment by dpvc** [comment]
I suspect that Peter may be right, that this is the bug with Edge where the `<math>` tags aren't being identified properly.  @iver56, could you check [this page](http://www.math.union.edu/locate/Cervone/transfer/mathjax/sample-mml.html) to see if it renders correctly?

It is unlikely that the problem is misidentifying the browser as Android or WebKit (though that may well be happening), because the issue seems to be in identifying the MathML, not displaying the results (since the results display correctly for TeX input).  The main differences between the browsers are in the output, not the input phase, so I would put my money on Peter's suggestion.

See if the page linked above works, and that may tell us more.

**Comment by iver56** [comment]
@dpvc That page does not render correctly on my Lumia 820 with Windows Phone 8.1

**Comment by Sebus13** [comment]
On my Lumia 820 with Windows Phone 8.1 it also doesn't render correctly:
![wp_ss_20150525_0001](https://cloud.githubusercontent.com/assets/11392432/7803136/5b7160fa-0348-11e5-9a91-98ca715ffbff.png)

**Comment by dpvc** [comment]
Thanks, @iver56 and @Sebus13.  I appreciate your willingness to do some testing for us.  I'll need to make up a more sophisticated test and see if we can't track down the issue.  I'll get back to you when I have the next test ready.

**Comment by dpvc** [comment]
OK, @iver56 and @Sebus13, I've put two new test files in place that should give me some information about how IE mobile is recognizing (or not) the math elements on the page.  Please try these two pages and let me know the output that you receive (this is diagnostic output, so it will not be rendered mathematics).

* [Test 1](http://www.math.union.edu/locate/Cervone/transfer/mathjax/IE-math-test.html)
* [Test 2](http://www.math.union.edu/locate/Cervone/transfer/mathjax/IE-math-test.xhtml)

**Comment by iver56** [comment]
Test 1:
![image](https://cloud.githubusercontent.com/assets/1470603/7816840/029bdb8e-03d1-11e5-8308-b0b9af03678c.png)

Test 2:
![image](https://cloud.githubusercontent.com/assets/1470603/7816842/074683b4-03d1-11e5-8bb0-22262b26f2c5.png)

**Comment by dpvc** [comment]
Thanks, @iver56, can you compare that to what you get in desktop mode?

**Comment by iver56** [comment]
@dpvc Here are the same two pages with the "website preference" setting set to "desktop version":

Test 1:
![image](https://cloud.githubusercontent.com/assets/1470603/7817276/38fec40e-03d4-11e5-8ddd-8a5b25d50f46.png)

Test 2:
![image](https://cloud.githubusercontent.com/assets/1470603/7817293/5a741396-03d4-11e5-937d-fb1d67997421.png)

**Comment by dpvc** [comment]
@iver56, thanks again.  The output for Test1 is incorrect, but consistent with IE9, so should have worked with MathJax.  Here is another test that actually does call MathJax, so give it a minute to finish up before capturing the output.  The final thing should be `Jax: n` where `n` is a number (3 one hopes).

* [Another Test](http://www.math.union.edu/locate/Cervone/transfer/mathjax/IE-math-test2.html)

**Comment by Sebus13** [comment]
Here are my screenshots:
mobile version:
![wp_ss_20150526_0006 1](https://cloud.githubusercontent.com/assets/11392432/7819151/21ef2ca2-03e0-11e5-91ff-0a847dc37cdf.png)
desktop version:
![wp_ss_20150526_0007 1](https://cloud.githubusercontent.com/assets/11392432/7819156/2e662af8-03e0-11e5-9ecf-5c0887140f74.png)

**Comment by dpvc** [comment]
@Sebus13, thanks, that confirms that the mml2jax component has located the math and set up the internal internal scripts for them.  So my initial idea of what was going wrong isn't correct, and it is something more complicated.  Now I'll have to try to make a test to figure out why they aren't getting processed.

**Comment by pkra** [comment]
@dpvc I'm wondering.... @Sebus13  @iver56 have ever you tried [VorlonJS](http://vorlonjs.com)?

**Comment by iver56** [comment]
I haven't tried it yet, but I'm willing to try it

**Comment by dpvc** [comment]
Well, from the results from @Sebus13, it looks like MathJax is finding and processing the math on the page (my original thought is that it wasn't doing so).  Now I am wondering if it is somehow selecting Native MathML output rather than HTML-CSS output.  I've put yet another test in place to see if that is what is happening.  Can @iver56 or @Sebus13 try out [this page](http://www.math.union.edu/locate/Cervone/transfer/mathjax/IE-math-test3.html) and post the output?  This may produce a lot of text, so if it is too much to capture, let me know and I'll try to filter it down.

**Comment by Sebus13** [comment]
I never tried VorlonJS. Here are the screenshots of last page:

mobile version:
![wp_ss_20150527_0001](https://cloud.githubusercontent.com/assets/11392432/7843670/6f43693e-04ab-11e5-9548-4a0447512898.png)

and desktop version:
![wp_ss_20150527_0002](https://cloud.githubusercontent.com/assets/11392432/7843682/821f4fa0-04ab-11e5-99da-013d470b2ab3.png)

**Comment by dpvc** [comment]
Great, thanks!  Thank confirms that the NativeMML output is being selected (incorrectly) for the mobile version.  I'll have to look into why that is happening.  

In the meantime, can you try [this test](http://www.math.union.edu/locate/Cervone/transfer/mathjax/IE-math-test4.html) that configures IE not to use NativeMML output?  I'm hoping that this will display properly in mobile mode.

**Comment by dpvc** [comment]
Oops, forgot to include @iver56 and @Sebus13 in my previous message.  Can one of you try out the test in the message above to see if it renders properly?  Thanks.

**Comment by Sebus13** [comment]
Yes it renders properly in mobile version:
![wp_ss_20150527_0003](https://cloud.githubusercontent.com/assets/11392432/7844871/8c0a5764-04b3-11e5-9969-974a721ad9d8.png)

**Comment by dpvc** [comment]
@Sebus13, thank again for the help.  So it seems that the automatic determination of whether to use NativeMML is not working for Mobile.  So perhaps @iver56 was right after all, and the browser detection is the culprit.

Can one of you run [this test](http://www.math.union.edu/locate/Cervone/transfer/mathjax/IE-math-test5.html) to find out what MathJax thinks IE mobile is?  Thanks again.

**Comment by dpvc** [comment]
PS, I've added the MatchWebFonts extension to see if that will improve the scaling.

**Comment by iver56** [comment]
@dpvc 
![image](https://cloud.githubusercontent.com/assets/1470603/7846608/0497ce18-04bf-11e5-9297-68997bf44137.png)

Apparently the problem is "Safari: true"

**Comment by pkra** [comment]
@dpvc try https://github.com/mathjax/MathJax-docs/issues/98 ?

**Comment by dpvc** [comment]
@iver56, yes, the `isSafari` is the culprit.  That is being set true due to the `AppleWebKit/537` in the user agent string.  I'll have to rule out the case where `like iPhone` is in the string as well.

@pkra, do we want a hot fix for this, or should it wait until v2.6?

It is certainly time to revisit the MMLorHTML.js file (which is where the NativeMML output is being selected).  Currently, MSIE is the only one that selects NativeMML output (when MathPlayer is present).  Perhaps we want HTML-CSS across the board (since MathPlayer is not able to work in current versions of IE).  And there is no current means of selecting SVG output (or the new CommonHTML output when it is available in 2.6).

**Comment by pkra** [comment]
@dpvc slight preference against another hotfix release (with June coming up etc). Let's discuss this F2F on Tuesday?

**Comment by dpvc** [comment]
=> Merged

**Comment by mostafiz93** [comment]
The problem still persists, this is the screenshot of http://mathjax.org in my lumia 820, internet explorer in mobile mode
![wp_ss_20150902_0001](https://cloud.githubusercontent.com/assets/1145582/9622083/9b5ce9d6-5150-11e5-90f4-21d80cdf9080.png)

**Comment by pkra** [comment]
@mostafiz93 this code is merged into develop but the release is not yet finished. You'll have to wait another week for the public beta to start (which mathjax.org traditionally switches to) or your could test the code locally (note: you'll need to load the unpacked version as packing only happens on release).
